src/ctim/schemas/weakness.cljc

```diff
@@ -0,0 +1,202 @@
+                              "that define this weakness"))
+   (f/entry :structure v/WeaknessStructure
+            :description (str "defines the structural nature of the weakness")))
+  (f/optional-entries
+   (f/entry :languages (f/set-of Language)
+            :description "Applicable Languages")
+   (f/entry :os (f/set-of OperatingSystem)
+            :description "Applicable operating systems")
+   (f/entry :architectures (f/set-of Architecture)
+            :description "Applicable architectures")
+   (f/entry :paradigms (f/set-of Paradigm)
+            :description "Applicable paradigms")
+   (f/entry :technologies (f/set-of Technology)
+            :description "Applicable technologies")
+   (f/entry :background_details c/Markdown
+            :description (str "information that is relevant but not related to "
+                              "the nature of the weakness itself"))
+   (f/entry :alternate_terms (f/set-of c/ShortString)
+            :description (str "indicates one or more other names used to "
+                              "describe this weakness"))
+   (f/entry :modes_of_introduction (f/set-of ModeOfIntroduction)
+            :description (str "information about how and when a given weakness "
+                              "may be introduced"))
+   (f/entry :likelihood v/HighMedLow
+            :description "Likelihood of exploit")
+   (f/entry :common_consequences (f/set-of Consequence)
+            :description (str "specify individual consequences associated "
+                              "with a weakness"))
+   (f/entry :detection_methods (f/set-of DetectionMethod)
+            :description (str "identify methods that may be employed to detect "
+                              "this weakness, including their strengths and "
+                              "limitations"))
+   (f/entry :potential_mitigations (f/set-of Mitigation)
+            :description (str "describe potential mitigations associated with "
+                              "a weakness"))
+   (f/entry :functional_areas (f/set-of v/FunctionalArea)
+            :description (str "identifies the functional area of the software "
+                              "in which the weakness is most likely to occur"))
+   (f/entry :affected_resources (f/set-of v/SystemResource)
+            :description (str "identify system resources that can be affected "
+                              "by an exploit of this weakness"))
+   (f/entry :notes c/Markdown
+            :description (str "provide any additional comments about the "
+                              "weakness"))))
+
+;; Does not contain these fields from CWE
+;; - Extended_Description
+;; - Related_Weaknesses (Relationship)
+;; - Weakness_Ordinalities
+;; - Demonstrative_Examples
+;; - Observed_Examples
+;; - Taxonomy_Mappings
+;; - Related_Attack_Patterns
+;; - References (external_references)
+;; - Content_History
+;; - Name
+;; - Abstraction
```

Comment:
We should support this as an optional field `:abstraction_level`, with the values enumeration of `["Class", "Base", "Variant",  "Compound"]` as defined https://cwe.mitre.org/documents/schema/#AbstractionEnumeration , similar to how we support it in Attack Patterns (https://github.com/threatgrid/ctim/blob/master/src/ctim/schemas/attack_pattern.cljc#L49)